lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -195,26 +231,57 @@ def initconfig_hardware_entries(self):
             "#------------------{0}\n".format("-" * 60),
         ]
 
+        # Provide standard CMake arguments for dependent CachedCMakePackages
         if spec.satisfies("^cuda"):
             entries.append("#------------------{0}".format("-" * 30))
             entries.append("# Cuda")
             entries.append("#------------------{0}\n".format("-" * 30))
 
             cudatoolkitdir = spec["cuda"].prefix
-            entries.append(cmake_cache_path("CUDA_TOOLKIT_ROOT_DIR", cudatoolkitdir))
-            cudacompiler = "${CUDA_TOOLKIT_ROOT_DIR}/bin/nvcc"
-            entries.append(cmake_cache_path("CMAKE_CUDA_COMPILER", cudacompiler))
+            entries.append(cmake_cache_path("CUDAToolkit_ROOT", cudatoolkitdir))
```

Comment:
Currently removing `CUDA_TOOLKIT_ROOT_DIR` would break any BLT packages since it partially still relies on the deprecated behavior. There is a PR to fix this but it isn't fully tested.

Reply:
The safest thing is to output both. But I can work around whatever is decided.

Reply:
So my packages don't need the deprecated variable, but I left it in for the BLT packages.  I would say that we leave it in for now.

Reply:
To be clear, do you mean you will add it back to this PR? I only see it being removed from this code block and don't see it added back anywhere else (and don't see any other references in `cmake.py` on the develop branch).

Reply:
@scheibelp @white238 I added it to the PR.